core/domain/feedback_services.py

```diff
@@ -165,6 +166,31 @@ def create_message(
 
     if author_id:
         subscription_services.subscribe_to_thread(author_id, full_thread_id)
+        update_messages_read_by_the_user(
+            exploration_id, thread_id, author_id, [message_id])
+
+
+def update_messages_read_by_the_user(exploration_id, thread_id,
+                                     user_id, message_ids):
+    """Adds the messages to the list of the messages read by the user.
+
+    Args:
+        exploration_id: str. The id of the exploration.
+        thread_id. str. The id of the thread.
+        user_id: str. The id of the user reading the messages,
+        message_ids: list(int): The ids of the messages to be added to the read
```

Comment:
In the implementation it seems like you are replacing this list, not adding to it...